### PR TITLE
[PULSEBS-11] As a booking manager I want to monitor usage (booking, cancellations, attendance) of the system

### PR DESCRIPTION
**Pull request by giovannitangredi**
Added routes and controllers for booking manager stats : 

- /api/stats/systemStats get the sum of booking, cancellations, attendance for all lectures
- /api/stats/lecturesStats get  a list of all lectures with booking, cancellations, attendance stats
- /api/courses/:courseid/stats get the sum of booking, cancellations, attendance for all lecturs of a specific course
- /api/courses/lecturesStats get  a list of all lectures of a specific course with booking, cancellations, attendance stats

### CODE REVIEW

File: server/controllers/usage-controller.js [hunk 1]
```diff
@@ -96,4 +96,82 @@ exports.getBookingStats = async (req, res) => {
         });
     }
   };
-  
\ No newline at end of file
+  
+  exports.getSystemStats = async (req,res)=>{
```
Comment (enricocarraro) [inline server/controllers/usage-controller.js]: Only managers should be able to access this data.

File: server/controllers/usage-controller.js [hunk 2]
```diff
@@ -96,4 +96,82 @@ exports.getBookingStats = async (req, res) => {
...
+      res.json(queryResults);
+    })
+    .catch((err)=>{
+      res.status(501).json({
+        message: `There was an error retrieving the system stats: ${err}`,
+      });
+    });
+  }
+
+  exports.getAllLecturesStats = async (req,res)=>{
```
Comment (enricocarraro) [inline server/controllers/usage-controller.js]: Only managers and teachers should be able to access this data.

File: server/controllers/usage-controller.js [hunk 3]
```diff
@@ -96,4 +96,82 @@ exports.getBookingStats = async (req, res) => {
...
+      res.json(queryResults);
+    })
+    .catch((err)=>{
+      res.status(501).json({
+        message: `There was an error retrieving the all the lectures stats: ${err}`,
+      });
+    });
+  }
+  
+  exports.getCourseTotalStats = async (req,res)=>{
```
Comment (enricocarraro) [inline server/controllers/usage-controller.js]: Same as above.

File: server/controllers/usage-controller.js [hunk 4]
```diff
@@ -96,4 +96,82 @@ exports.getBookingStats = async (req, res) => {
...
+      res.json(queryResults);
+    })
+    .catch((err)=>{
+      res.status(501).json({
+        message: `There was an error retrieving the course  stats: ${err}`,
+      });
+    });
+  }
+
+  exports.getCourseLecturesStats = async (req,res)=>{
```
Comment (enricocarraro) [inline server/controllers/usage-controller.js]: Same as above.

File: server/routes/stats-route.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// Import express
+const express = require("express");
+
+// Import usage-controller
+const usageRoutes = require("../controllers/usage-controller");
```
Comment (enricocarraro) [inline server/routes/stats-route.js]: I would prefer naming consistency here, just choose one: either `usage` or `stats`.

File: server/routes/stats-route.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+// Create router
+const router = express.Router();
+
+router.get("/systemStats", usageRoutes.getSystemStats);
```
Comment (enricocarraro) [inline server/routes/stats-route.js]: `"/systemStats"`-> `"/system"`
We are in the `stats` route, we don't need to specify the obvious.

File: server/routes/stats-route.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+router.get("/lecturesStats", usageRoutes.getAllLecturesStats);
```
Comment (enricocarraro) [inline server/routes/stats-route.js]: As above.

File: server/routes/course-route.js [hunk 8]
```diff
@@ -12,6 +12,8 @@ const router = express.Router();
 router.get("/", courseRoutes.getCourses);
 router.get("/:courseid/lectures", lectureRoutes.getScheduledLectures);
 router.get("/:courseid/bookings", usageRoutes.getBookingStats);
+router.get("/:courseid/stats", usageRoutes.getCourseTotalStats);
```
Comment (enricocarraro) [inline server/routes/course-route.js:15]: I think this route are not very well suited for the "course" route, they would make more sense under the "stats" route (e.g. `(stats)/:courseid`).

File: server/routes/course-route.js [hunk 9]
```diff
@@ -12,6 +12,8 @@ const router = express.Router();
...
+router.get("/:courseid/lecturesStats", usageRoutes.getCourseLecturesStats);
```
Comment (enricocarraro) [inline server/routes/course-route.js:16]: As above.

File: server/controllers/usage-controller.js (hunk 1 above)
Comment (giovannitangredi) [reply server/controllers/usage-controller.js]: ok, now it sent an error message if anyother role try to access to the queries

**Pull request closed without merging**